bson/src/main/org/bson/json/JsonReader.java

```diff
@@ -1211,6 +1216,17 @@ private int readIntFromExtendedJson() {
         return value;
     }
 
+    private BsonBinary visitUuidExtendedJson() {
+        verifyToken(JsonTokenType.COLON);
+        String uuidString = readStringFromExtendedJson();
+        verifyToken(JsonTokenType.END_OBJECT);
+        try {
+            return new BsonBinary(UUID.fromString(uuidString));
+        } catch (IllegalArgumentException e) {
+            throw new JsonParseException(e);
```

Comment:
I read about this "exception translation" technique in Effective Java :)

Reply:
Nice!